### PR TITLE
scheduler: rename scatter-range to scatter-range-scheduler

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/8379

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/http/api.go [hunk 1]
```diff
@@ -182,11 +181,6 @@ func SchedulerByName(name string) string {
 	return fmt.Sprintf("%s/%s", Schedulers, name)
 }
 
-// ScatterRangeSchedulerWithName returns the scatter range scheduler API with name parameter.
-func ScatterRangeSchedulerWithName(name string) string {
```
Comment (rleungx) [inline client/http/api.go:186]: It's a break change.

Comment (okJiang) [reply client/http/api.go:186]: It is not used🤔

Comment (rleungx) [reply client/http/api.go:186]: TiDB uses it.

Comment (okJiang) [reply client/http/api.go:186]: I have not found it in TiDB. And where is the service interface implement in PD server? 🤔

Comment (rleungx) [reply client/http/api.go:186]: https://github.com/pingcap/tidb/blob/2a3352c45dd0f8dd5102adb92879bbfa964e7f5f/pkg/server/handler/tikvhandler/tikv_handler.go#L1252

File: tools/pd-ctl/pdctl/command/scheduler.go [hunk 2]
```diff
@@ -423,6 +424,17 @@ func addSchedulerCommandFunc(cmd *cobra.Command, args []string) {
 func NewScatterRangeSchedulerCommand() *cobra.Command {
 	c := &cobra.Command{
 		Use:   "scatter-range [--format=raw|encode|hex] <start_key> <end_key> <range_name>",
+		Short: "scatter-range will be deprecated in the future, please use scatter-range-scheduler instead",
+		Run:   addSchedulerForScatterRangeCommandFunc,
+	}
+	c.Flags().String("format", "hex", "the key format")
```
Comment (HuSharp) [inline tools/pd-ctl/pdctl/command/scheduler.go:430]: When will this command be deprecated?
And maybe you can add information in `Deprecated`
ref: https://github.com/spf13/cobra/blob/main/command.go#L102

Comment (okJiang) [reply tools/pd-ctl/pdctl/command/scheduler.go:430]: The exact point of deprecation still needs to be discussed with the PM, no conclusion for now.

File: client/http/api.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ const (
 	RegionLabelRulesByIDs = "/pd/api/v1/config/region-label/rules/ids"
 	// Scheduler
 	Schedulers            = "/pd/api/v1/schedulers"
-	scatterRangeScheduler = "/pd/api/v1/schedulers/scatter-range-"
+	scatterRangeScheduler = "/pd/api/v1/schedulers/scatter-range-scheduler-"
```
Comment (okJiang) [inline client/http/api.go:63]: need update tidb repo

File: tools/pd-ctl/pdctl/command/scheduler.go [hunk 4]
```diff
@@ -422,7 +423,19 @@ func addSchedulerCommandFunc(cmd *cobra.Command, args []string) {
 // NewScatterRangeSchedulerCommand returns a command to add a scatter-range-scheduler.
 func NewScatterRangeSchedulerCommand() *cobra.Command {
 	c := &cobra.Command{
-		Use:   "scatter-range [--format=raw|encode|hex] <start_key> <end_key> <range_name>",
+		Use:        "scatter-range [--format=raw|encode|hex] <start_key> <end_key> <range_name>",
+		Short:      "scatter-range will be deprecated in the future, please use scatter-range-scheduler instead",
+		Run:        addSchedulerForScatterRangeCommandFunc,
```
Comment (HuSharp) [inline tools/pd-ctl/pdctl/command/scheduler.go:428]: Can we remove `short` and `run` directly?

Comment (okJiang) [reply tools/pd-ctl/pdctl/command/scheduler.go:428]: Generally, the deprecation process is gradual, so I'm still keeping it and then removing it completely after a period of time (the exact time is still up for discussion)

Comment (HuSharp) [reply tools/pd-ctl/pdctl/command/scheduler.go:428]: Got it, make sense to `run`.
When `Deprecated` is added, the command will be considered `unavailable`. This means this command will not appear in `help`, so maybe we can remove `short` directly?

Comment (okJiang) [reply tools/pd-ctl/pdctl/command/scheduler.go:428]: Oh, I don't know that. I will do a test for it and choose a best practice.

Comment (okJiang) [reply tools/pd-ctl/pdctl/command/scheduler.go]: `Short` has been removed. Thanks for your suggestion~

**Review by HuSharp — APPROVED** [review]

File: pkg/schedule/schedulers/scatter_range.go [hunk 5]
```diff
@@ -90,7 +90,7 @@ func (conf *scatterRangeSchedulerConfig) getEndKey() []byte {
 func (conf *scatterRangeSchedulerConfig) getSchedulerName() string {
 	conf.RLock()
 	defer conf.RUnlock()
-	return fmt.Sprintf("scatter-range-%s", conf.RangeName)
+	return fmt.Sprintf("%s-%s", types.ScatterRangeScheduler, conf.RangeName)
```
Comment (rleungx) [inline pkg/schedule/schedulers/scatter_range.go:93]: need to consider the compatibility issue

Comment (okJiang) [reply pkg/schedule/schedulers/scatter_range.go:93]: https://github.com/tikv/pd/pull/8607/commits/7f1434335bb5f623cc8aa6041aabb45eaee48be7

This commit fixed it. You can check it in its comment.

Comment (okJiang) [reply pkg/schedule/schedulers/scatter_range.go:93]: friendly ping

**Review by rleungx — APPROVED** [review]

**Pull request merged**